### PR TITLE
Add rollup, eslint, and matches polyfill

### LINKED ISSUE #51: Exception in Edge: Object doesn't support property or method 'matches'
I keep getting this exception in Edge, line inert.js (174,9) - this happens also in the demo page and the checkboxes don't seem to work.

OS: Windows 10, 
Browser:Microsoft Edge 38.14393.0.0

### PR DESCRIPTION
**Pull request by robdodson**
Fixes #51

Kind of a big PR, sorry about that.

This switches our build over to using [Rollup](https://rollupjs.org/) instead of Gulp. That means we can easily work with ES modules now (if we wanted to break the polyfill into smaller modules we could).

I'm also importing the dom-matches polyfill for IE/Edge support and producing a UMD build which will work standalone or with module loaders like browserify.

Also added ESLint so we could have more consistent PRs. I'm not currently linting the test directory because there were a handful of issues with mocha I didn't feel like resolving at the moment. Kinda want to land this PR first :)

I also killed the `inert.min.js` build for now because Uglify has issues with module syntax. There's other tools that will do the right thing but I wanted to try doing them in a different PR.

@alice PTAL
@vcorr FYI

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      "es2015",
+      {
+        "modules": false
+      }
+    ]
+  ],
+  "plugins": [
+    "external-helpers"
+  ]
+}
```
Comment (alice) [inline .babelrc]: newline?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+rollup.config.js
+dist/**/*.js
+node_modules/**/*.js
```
Comment (alice) [inline .eslintignore]: newline? I'm seeing this a lot lately - does VS Code strip EOF newlines by default or something?

File: src/inert.js [hunk 3]
```diff
@@ -72,10 +73,10 @@ class InertRoot {
     // Watch for:
     // - any additions in the subtree: make them unfocusable too
     // - any removals from the subtree: remove them from this inert root's managed nodes
-    // - attribute changes: if `tabindex` is added, or removed from an intrinsically focusable element,
-    //   make that node a managed node.
+    // - attribute changes: if `tabindex` is added, or removed from an intrinsically focusable
+    //   element, make that node a managed node.
     this._observer = new MutationObserver(this._onMutation.bind(this));
-    this._observer.observe(this._rootElement, { attributes: true, childList: true, subtree: true });
```
Comment (alice) [inline src/inert.js]: Why remove these spaces? Does ESLint insist on it? Is there an option to make it not do that?

File: src/inert.js [hunk 4]
```diff
@@ -109,7 +110,9 @@ class InertRoot {
    * @param {Node} startNode
    */
   _makeSubtreeUnfocusable(startNode) {
-    composedTreeWalk(startNode, (node) => { this._visitNode(node); });
+    composedTreeWalk(startNode, (node) => {
+      this._visitNode(node);
```
Comment (alice) [inline src/inert.js]: Hm, remove {}s instead of spreading out over 3 lines?

File: src/inert.js [hunk 5]
```diff
@@ -170,7 +173,9 @@ class InertRoot {
    * @param {Node} startNode
    */
   _unmanageSubtree(startNode) {
-    composedTreeWalk(startNode, (node) => { this._unmanageNode(node); });
+    composedTreeWalk(startNode, (node) => {
```
Comment (alice) [inline src/inert.js]: Similarly, remove {}s here?

File: src/inert.js [hunk 6]
```diff
@@ -298,9 +303,12 @@ class InertNode {
     return this._destroyed;
   }
 
+  /**
+   * Throw if user tries to access destroyed InertNode.
+   */
   _throwIfDestroyed() {
     if (this.destroyed)
-      throw new Error("Trying to access destroyed InertNode");
+      throw new Error('Trying to access destroyed InertNode');
```
Comment (alice) [inline src/inert.js]: Oh no, can we go back to double quotes?

File: src/inert.js [hunk 7]
```diff
@@ -298,9 +303,12 @@ class InertNode {
     return this._destroyed;
   }
 
+  /**
+   * Throw if user tries to access destroyed InertNode.
```
Comment (alice) [inline src/inert.js]: Isn't this fairly redundant with the method name?

File: .eslintignore (hunk 2 above)
Comment (robdodson) [inline .eslintignore]: I'm not sure. Maybe there's a setting I can enable. I'll look

File: src/inert.js (hunk 3 above)
Comment (robdodson) [inline src/inert.js]: It's part of the Google ESLint rules:
https://github.com/google/eslint-config-google/blob/master/index.js#L246

"off" or 0 - turn the rule off
"warn" or 1 - turn the rule on as a warning (doesn’t affect exit code)
"error" or 2 - turn the rule on as an error (exit code is 1 when triggered)

File: src/inert.js (hunk 7 above)
Comment (robdodson) [inline src/inert.js]: yes. the ESLint rules require a JSDoc comment so I just added one in haste.

File: src/inert.js (hunk 6 above)
Comment (robdodson) [inline src/inert.js]: gross but ok :)

Comment (alice) [inline src/inert.js]: Ok let's leave it with single quotes.

**Comment by vcorr** [comment]
The README still refers to inert.min.js
`The dist directory contains two versions of the polyfill. inert.js is the debug version, containing an inlined sourcemap, inert.min.js is the production version, minified to be as small as possible.`

**Comment by robdodson** [comment]
good catch, thank you

**Pull request merged**